program/src/processor.rs

```diff
@@ -505,13 +512,14 @@ pub fn process_unstake(
             );
             return Err(LidoError::InvalidAmount.into());
         }
-        validator.entry.stake_seeds.begin += 1;
+        validator.stake_seeds.begin += 1;
     }
 
-    validator.entry.unstake_accounts_balance = (validator.entry.unstake_accounts_balance + amount)?;
-    validator.entry.unstake_seeds.end += 1;
+    validator.unstake_accounts_balance = (validator.unstake_accounts_balance + amount)?;
+    validator.effective_stake_balance = validator.get_effective_stake_balance();
```

Comment:
Also here, it seems safer if it would be a method `recompute_effective_stake_balance()`, also so you cannot assign it to the wrong field or forget to assign it.

Reply:
Same here, answered earlier. To be considered...

Reply:
Maybe I should rename `get_effective_stake_balance` to `recompute_effective_stake_balance` that does not mutate anything?

```rust
    pub fn recompute_effective_stake_balance(&self) -> Lamports {
        (self.stake_accounts_balance - self.unstake_accounts_balance)
            .expect("Unstake balance cannot exceed the validator's total stake balance.")
    }
```